### PR TITLE
Fixes to make training work with new dependencies

### PR DESCRIPTION
**Pull request by izderadicka**
I have installed freshly from repo, and hit issues (exceptions), when I wanted to train model with new data (Czech language - from https://huggingface.co/datasets/Thomcles/Czech-Speech-Monospeaker-Honza,  actually data are not good, had to do some cleanup,  but anyhow it was learning exercise) .  Problems I encountered:

- Was not able to use existing checkpoints (jirka or cori)  - first problem was with globals in model, second with extra model parameters, which pytorch is now checking.
- Second problem was with export -  again parsing some extra arguments and also with dynamo as default now.

Below are fixes, which I've done with help of Claude Code, as I'm not an expert in pytorch.   These helped me to train model to first checkpoint and export checkpoint to onnx and test onnx (quality was poor, but that I expect having only one epoch of training).   I was testing on my notebook without NVidia card - just to test the process. So I cannot guarantee that my fixes works flawlessly -  when exporting module got some warnings like:

```
/home/ivan/workspace/piper/piper1-gpl/src/piper/train/vits/transforms.py:174: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  assert (discriminant >= 0).all(), discriminant
/home/ivan/workspace/piper/.venv/lib/python3.12/site-packages/torch/onnx/_internal/torchscript_exporter/jit_utils.py:308: UserWarning: Constant folding - Only steps=1 can be constant folded for opset >= 10 onnx::Slice op. Constant folding not applied. (Triggered internally at /pytorch/torch/csrc/jit/passes/onnx/constant_fold.cpp:178.)
```

### CODE REVIEW

**Comment by synesthesiam** [comment]
## Review

Thanks for this — the motivation is real and the torch-2.9 diagnosis is accurate (this env runs torch 2.9.1, pin is `torch>=2,<3`). A few things need addressing before merge.

### Blocking: `_parse_ckpt_path` is never called

`_parse_ckpt_path` is defined on `VitsLightningCLI` but there's no call site — LightningCLI does not auto-invoke a method by that name. To run before class instantiation it needs to override a real hook, e.g.:

```python
def before_instantiate_classes(self) -> None:
    self._parse_ckpt_path()
```

As written, the checkpoint hparam-filtering logic (the bulk of the additions) never executes, so the *"extra model parameters which pytorch is now checking"* part of the problem isn't actually fixed by this diff — only the `add_safe_globals` line does anything in `__main__.py`.

### Staleness against `main`

- **`dataset.py`**: `main` already fixed this — `train_dataloader` now calls `self._make_dataloader(self.train_dataset, shuffle=True, drop_last=True)` (with an explanatory comment), so this hunk is redundant and will conflict.
- **`__main__.py`**: `main` has since added `ModelCheckpoint`/`_DEFAULT_CALLBACKS` and changed the imports, so this hunk won't apply cleanly either.

### The parts that are correct and worth keeping

- **`dynamo=False`** in `torch.onnx.export` — correct and necessary. torch ≥2.9 defaults `dynamo=True`, and the spline transforms use data-dependent control flow the dynamo exporter can't trace. Current `main` still lacks this, so it's a genuine gap.
- **`_instantiator=None`** in `load_from_checkpoint` — reasonable; `VitsModel.__init__` accepts `**kwargs`, so unknown hparams are harmless when instantiated directly. Well-commented.
- **`add_safe_globals([pathlib.PosixPath])`** — legitimate fix for `weights_only` unpickling of Linux-saved checkpoints.

### Minor notes

- The `export_onnx.py` comments are great — they explain *why*, which is exactly right for version-sensitive workarounds.
- Once wired, `torch.load(..., weights_only=True)` in `_parse_ckpt_path` will still trip on any custom global beyond `PosixPath` (e.g. a `WindowsPath` checkpoint) — worth a comment.
- For consistency, prefer `_LOGGER.error(...)` over `sys.stderr.write(...)` in the failure branch.

### Suggested path

1. **Split** the export fixes (`dynamo=False` + `_instantiator=None`) into their own PR against current `main` — clean, correct, still missing upstream.
2. **Drop** the `dataset.py` hunk (already fixed in `main`).
3. **Rework** the checkpoint-hparam fix: wire `_parse_ckpt_path` into `before_instantiate_classes`, rebase onto the current `__main__.py`, and consider whether much of it is still needed now that `VitsModel` accepts `**kwargs` and dedicated warmstart flags exist.

*Reviewed with Claude Code.*

**Comment by izderadicka** [comment]
Thanks, vibed code vs robot's review - these are today's realities - I ll try to rebase latest main  - to see what will remain ...
